services/watson-extension/src/watson_extension/routes/platform/rbac.py

```diff
@@ -0,0 +1,48 @@
+import injector
+from pydantic import BaseModel
+
+from quart import Blueprint, render_template
+from quart_schema import validate_response, validate_querystring, document_headers
+
+from watson_extension.core.platform.rbac import (
+    RBACCore,
+)
+from watson_extension.routes import RHSessionIdHeader
+
+blueprint = Blueprint("rbac", __name__, url_prefix="/rbac")
+
+
+class TamAccessRequestQuery(BaseModel):
+    account_id: str
+    org_id: str
+    duration: str
+
+
+class TamAccessRequestResponse(BaseModel):
+    response: str
+
+
+@blueprint.post("/tam-access")
+@validate_querystring(TamAccessRequestQuery)
+@validate_response(TamAccessRequestResponse)
+@document_headers(RHSessionIdHeader)
+async def send_tam_access(
+    query_args: TamAccessRequestQuery,
+    rbac_core: injector.Inject[RBACCore],
+) -> TamAccessRequestResponse:
+    start_date, end_date = rbac_core.get_start_end_date_from_duration(
+        query_args.duration
+    )
+    roles = await rbac_core.get_roles_for_tam()
+    ok = await rbac_core.send_rbac_tam_request(
+        query_args.account_id, query_args.org_id, start_date, end_date, roles
+    )
+
+    # TODO check if they are internal, need a helper?
+    return TamAccessRequestResponse(
+        response=await render_template(
+            "platform/rbac/tam_access_request.txt.jinja",
```

Comment:
Is this portion being worked on currently?

Reply:
*Now* it is done haha, ty!